index.js

```diff
@@ -3983,6 +3983,26 @@
   S.splitOn =
   def('splitOn', {}, [$.String, $.String, $.Array($.String)], splitOn);
 
+  //# splitOnRegex :: RegExp -> String -> Array String
+  //.
+  //. Returns the substrings of its second argument separated by evaluated
+  //. matches of its first argument.
+  //.
+  //. See also [`splitOn`](#splitOn).
+  //.
+  //. ```javascript
+  //. > S.splitOnRegex(/\s+/, 'foo  bar baz   qux')
+  //. ['foo', 'bar', 'baz', 'qux']
+  //. ```
+  function splitOnRegex(pattern, s) {
+    return s.split(pattern);
+  }
+  S.splitOnRegex =
+  def('splitOnRegex',
+    {},
+    [$.RegExp, $.String, $.Array($.String)],
+    splitOnRegex);
```

Comment:
I considered this especially since ramda has it in a single function as well. I just assumed that this decision had already been made because of the wording of the related issue and didn't want to step any anybody's toes with my first contribution 😀

Happy to change it to fit this.